metricbeat/mb/module/wrapper.go

```diff
@@ -298,8 +304,33 @@ func (msw *metricSetWrapper) String() string {
 		msw.module.Name(), msw.Name(), msw.Host())
 }
 
+func (msw *metricSetWrapper) Test(d testing.Driver) {
+	done := make(chan struct{})
+	d.Run(msw.Name(), func(d testing.Driver) {
+		// ReportingMetricSet would hang out forever, perhaps we can add a timeout based test in the future
+		if _, ok := msw.MetricSet.(mb.ReportingMetricSet); ok {
+			d.Warn("test", "metricset doesn't support testing")
+			return
+		}
+
+		reporter := &testingReporter{
+			driver: d,
+			done:   done,
+		}
+		msw.fetch(reporter)
+	})
+}
+
 // Reporter implementation
 
+type reporter interface {
+	StartFetchTimer()
+	Done() <-chan struct{}
```

Comment:
Would it make sense to embed `mb.PushReporter` in `reporter` rather than duplicating the method signatures?